### ISSUE TITLE
Miner efficiency

### ISSUE THREAD

**Issue by jordansafer** [jordansafer/screeps#38]
TLDR bulky miners are only worth it in PC rooms. Savings is 10% cpu for the city, cost is 2% of city energy production***. Total savings is 5% of empire cpu.

***(4k for miners + 1k for tug - 4k energy recovered from spawn gap).

Miners currently use 20% of city cpu (additional 20% used for power mining, and 20% used for transporters)
In a room w/o a pc, a miner mines with 20 work, additional cost is 2k*2 miners = 4k energy for 40 work and 10% cpu saving (20 move tugboat [1k] code required). production is 3k * 2 sources * 5 cycles = 30k in that time.

If we boost a miner to 40 work, we are spending 4k+/30k (10-15% of energy) for 10% cpu. Not worth it.

In a PC room, each source has an extra 100*250 = 25k energy, for a total of 80k energy for the room. In these rooms the change may be worth it since we save 10% CPU for a 5-6% energy cost.

PC room cpu numbers:
PC rooms make 160k/310k per 1.5k ticks, over 50% of miner cpu. Saving 50% of pc miner cpu saves 25% of empire miner cpu => 5% of city cpu empire-wide.

Numbers to remember:
energy production: 30k/1500 ticks
pc room production: 80k/1500 ticks
current miner cost (2): 6k/1500 ticks (1-2% lost- walking time, 7% lost- spawn time**), 
current transporter cost (2): 3k/1500 ticks
ferry (1): 1.5k/1500

**miner mines 40 energy/tick. when a miner dies, 150 ticks are lost worst case, so a miner has 150 ticks to mine 3k energy. 150*40 = 6k, 0 lost. In a PC room, there is 5k extra energy, so 2k is lost per miner. So a PC room loses 4k/80k = 5%. Currently, with 2 PC rooms in the empire, total production is 5*30k + 2*80k = 310k. Current losses from this issue are 8k/310k ~= 3%. (issue can also be prevented with bulky miner)

![Screen Shot 2019-12-11 at 12 03 55 AM](https://user-images.githubusercontent.com/8800717/70677766-1a2ece00-1c5e-11ea-8456-3b348219712b.png)
